### ISSUE TITLE
[Concurrency] ReferenceWritableKeyPath of main-actor bound property cannot be created inside a non-isolated context

### ISSUE THREAD

**Issue by xavierjurado** [swiftlang/swift#77774]
### Description

While using a ReferenceWritableKeyPath should be limited to the actor where the property it's referencing is bound, creating such key path should be allowed on any context.

### Reproduction

```swift
  nonisolated func testKeyPath() {
    let keyPath: ReferenceWritableKeyPath<UIView, _> = \.backgroundColor // Error: Cannot form key path to main actor-isolated property 'backgroundColor'
  }
``` 

### Expected behavior

`ReferenceWritableKeyPath` can be created from any context, but the compiler prevents applying such keypath outside of the context where the property is bound (ie: outside of `@MainActor` in the example above).

### Environment

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)

### Additional information

cc @hborla

**Comment by hborla** [comment]
I agree that there should be a way to express this. To allow forming isolated key paths from outside the isolation domains, the compiler needs to preserve the type of the global actor statically in order to check that type against the current isolation at the point of application. This isn't something that can be expressed for key path types today. The only way to represent this currently is using a main-actor-isolated function instead of a key-path.
